### PR TITLE
Update changelog for release (v0.17.1)

### PR DESCRIPTION
**Pull request by mccurdyc**
:star2:  [Rendered CHANGELOG](https://github.com/terraform-providers/terraform-provider-fastly/pull/270/commits/2ad2dad4685bff94766f2cacd7565546e686f267?short_path=4ac32a7#diff-4ac32a78649ca5bdd8e0ba38b7006a1e)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,11 @@
 ## 0.18.0 (Unreleased)
+
+## 0.17.1 (June 24, 2020)
```
Comment (phamann) [inline CHANGELOG.md]:
```suggestion
## 0.17.1 (Unreleased)
```

**Review by phamann — CHANGES_REQUESTED** [review]
LGTM other than date issue

File: CHANGELOG.md (hunk 1 above)
Comment (mccurdyc) [reply CHANGELOG.md]: Ah okay so that has to be Unreleased. Until the release is actually cut.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,11 @@
 ## 0.18.0 (Unreleased)
+
```
Comment (mccurdyc) [inline CHANGELOG.md]: @phamann do we need this `0.18.0` at the top still?

Comment (phamann) [reply CHANGELOG.md:2]: Yep.

Comment (phamann) [reply CHANGELOG.md]: Nope, looks like we don't: https://github.com/terraform-providers/terraform-provider-fastly/commit/820b30dd51e94bcb106a0b48938b96ac430ecbe8

Comment (phamann) [inline CHANGELOG.md]:
```suggestion
```

**Review by phamann — CHANGES_REQUESTED** [review]
LGTM other than your comment

**Comment by mccurdyc** [comment]
Force-pushed squashed commits addressing review comments.

**Review by phamann — APPROVED** [review]
LGTM :+1:

**Pull request merged**